addons/mail/models/models.py

```diff
@@ -31,3 +34,123 @@ def _notify_email_header_dict(self):
+            which makes code easier to extend if other failure messages added.
+            * 'failure_type' strings follow the structure : [MODULE_NAME__TARGET_FAILURE]
+
+            Failure messages can be added/overrided in other modules using 'module_messages' param.
+        """
+        messages = {
+            "mail__smtp_connection": lambda failed_emails_counter, mail_sms_notifications_window_action: (
+                'Odoo has been unable to send email(s)',
+                """%d email(s) could not be sent due to technical issues.<br/>
+                To solve SMTP connection related issues make sure that SMTP configuration is correct and click on "Test Connection", 
+                if your configuration is incorrect, an error message should help you diagnose the issue.<br/>
+                Ex : If you get a <b>[AUTHENTICATIONFAILED] Invalid credentials (Failure) warning</b> when you 
+                Test Connection on a Gmail address, activate the Less secure app access option.<br/>
+                In addition to that, enable the IMAP setting on your Gmail account.<br/>
+                You can check your mail <a href="/web#action=%s&amp;view_type=list">notifications</a> for more details about 
+                failed emails (an error message should help you diagnose the issue.)
+                """ % (failed_emails_counter, mail_sms_notifications_window_action),
+                'https://www.odoo.com/documentation/user/13.0/discuss/email_servers.html'
+            ),
+
+            "mail__invalid_recipient": lambda failed_ignored_emails_counter, mail_sms_notifications_window_action: (
+                'failed emails due to invalid email address',
+                """%d email(s) failed / ignored to avoid blocking delivery to other recipients.<br/>
+                Please make sure that recipient's email address really exists and if it's actually valid.<br/>
+                You can check your mail <a href="/web#action=%s&amp;view_type=list">notifications</a> for more details about 
+                failed emails (an error message should help you diagnose the issue.)
+                """ % (failed_ignored_emails_counter, mail_sms_notifications_window_action)
+            ),
+
+            "iap__low_credits": lambda account_credits: (
+                'Your credits balance is too low',
+                """Your current balance reached <b>%1.2f</b>, It’s time to recharge your credits!<br/>
+                To avoid service interruption (SMS, snailmail, lead enrichment...),
+                please go to your "IAP Portal" <b>[Settings app > Odoo IAP > View my Services]</b>.<br/>
+                From there, you can view your current balance, recharge your credits, review your consumption and set another 
+                reminder (by email) to when credits are low.""" % (account_credits),
+                'https://www.odoo.com/documentation/user/13.0/general/in_app_purchase/in_app_purchase.html'
```

Comment:
You can re-read your diff, and replace the documentation links from ``13.0`` to ``online``.